src/azure-cli/azure/cli/command_modules/storage/operations/account.py

```diff
@@ -522,17 +522,36 @@ def add_network_rule(cmd, client, resource_group_name, account_name, action='All
 
 def remove_network_rule(cmd, client, resource_group_name, account_name, ip_address=None, subnet=None,
                         vnet_name=None, tenant_id=None, resource_id=None):  # pylint: disable=unused-argument
+    from azure.cli.core.azclierror import InvalidArgumentValueError
+    from knack.util import todict
     sa = client.get_properties(resource_group_name, account_name)
     rules = sa.network_rule_set
     if subnet:
+        if all(not x.virtual_network_resource_id.endswith(subnet) for x in rules.virtual_network_rules):
+            raise InvalidArgumentValueError(
+                "Specified subnet {} is not in the storage account network rules. Found subnets are {}. Note: "
+                "resource id is case sensitive.".format(
+                    subnet, [x.virtual_network_resource_id for x in rules.virtual_network_rules]))
         rules.virtual_network_rules = [x for x in rules.virtual_network_rules
                                        if not x.virtual_network_resource_id.endswith(subnet)]
     if ip_address:
+        IpRule = cmd.get_models('IPRule')
+        ip_rule = IpRule(ip_address_or_range=ip_address)
+        if ip_rule not in rules.ip_rules:
+            raise InvalidArgumentValueError(
+                "Specified ip address {} is not in the storage account network rules. Found ip addresses are {}.".
+                format(ip_address, [x.ip_address_or_range for x in rules.ip_rules]))
         rules.ip_rules = [x for x in rules.ip_rules if x.ip_address_or_range != ip_address]
 
     if resource_id:
-        rules.resource_access_rules = [x for x in rules.resource_access_rules if
-                                       not (x.tenant_id == tenant_id and x.resource_id == resource_id)]
+        ResourceAccessRule = cmd.get_models('ResourceAccessRule')
+        resource_access_rule = ResourceAccessRule(resource_id=resource_id, tenant_id=tenant_id)
+        if resource_access_rule not in rules.resource_access_rules:
+            raise InvalidArgumentValueError(
+                "Specified resource access rule with resource id {} and tenant id {} is not in the storage account "
+                "network rules. Found resource access rules are {}. Note: resource id is case sensitive.".format(
+                    resource_id, tenant_id, [todict(x) for x in rules.resource_access_rules]))
+        rules.resource_access_rules = [x for x in rules.resource_access_rules if x != resource_access_rule]
```

Comment:
Can we compare the length after filter with the origin length to avoid one more traverse?

Just small suggestion :)

Reply:
good point. will update it with new method.